src/journal/journald-stream.c

```diff
@@ -461,9 +508,28 @@ static int stdout_stream_process(sd_event_source *es, int fd, uint32_t revents,
                 goto terminate;
         }
 
-        l = read(s->fd, s->buffer+s->length, sizeof(s->buffer)-1-s->length);
-        if (l < 0) {
+        /* If the buffer is full already (discounting the extra NUL we need), add room for another 1K */
+        if (s->length + 1 >= s->allocated) {
+                size_t new_size;
 
+                if (s->length + 1 + 1024 < s->length) { /* overflow? */
+                        assert(s->server->line_max > s->length);
+                        new_size = s->server->line_max + 1;
+                } else
+                        new_size = s->length + 1 + 1024;
```

Comment:
I don't think the overflow could ever happen. `length` is size_t, and `line_max` is limited to `SSIZE_T_MAX`.